adapters/saml/core/src/main/java/org/keycloak/adapters/saml/config/parsers/DeploymentBuilder.java

```diff
@@ -214,6 +220,25 @@ public SamlDeployment build(InputStream xml, ResourceLoader resourceLoader) thro
         return deployment;
     }
 
+    private int convertClockSkewInMillis(int duration, TimeUnit unit) {
```

Comment:
I'd suggest to just cut the rest from nanoseconds and microsecods, It'd simplify the method. We also doesn't need `long` here, `int` would be sufficient. And the INFO log message with value in ms should be added.
```suggestion
    private int convertClockSkewInMillis(int duration, TimeUnit unit) {
        int durationMillis = (int) unit.toMillis(duration);
        switch (unit) {
            case NANOSECONDS:
            case MICROSECONDS:
                log.warn("Clock skew value will be rounded down.");
            default:
                log.info("Clock skew set to " + durationMillis + "ms.");
        }
        return durationMillis;
    }
```